lib/lints/proper_super_calls/proper_super_calls_rule.dart

```diff
@@ -56,27 +57,38 @@ class ProperSuperCallsRule extends SolidLintRule {
         if (methodName == _initState || methodName == _dispose) {
           final statements = (node.body as BlockFunctionBody).block.statements;
 
-          checkSuperCalls(node, methodName, statements, reporter);
+          final hasOverrideAnnotation = node.metadata
+              .any((annotation) => annotation.name.name == _override);
```

Comment:
Probably best to move this inside the `_checkSuperCalls` method.

Reply:
Maybe even worth checking out the rest of the code -- maybe we have the logic of this check duplicated somewhere. If so -- move it to common utils.